### ISSUE TITLE
jupyter notebook kernel crashes when doing groupby, having one row with NaN

### ISSUE THREAD

**Issue by rojour** [pandas-dev/pandas#15791]
#### Code Sample, a copy-pastable example if possible

```python

df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)

```
#### Problem description

Data set has 26630 rows and one row of df["City"] has a NaN. The jupyter notebook kernel crashes.

#### Expected Output

A list of the top 20 cities is expected here.

df[df['City'].notnull()].groupby('City').count()['Bib'].sort_values(ascending=False).head(20)

produces the expected output.

https://github.com/rojour/boston_results

#### Output of ``pd.show_versions()``

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Darwin
OS-release: 16.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.2
nose: 1.3.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.11.3
scipy: 0.18.1
statsmodels: 0.6.1
xarray: None
IPython: 5.1.0
sphinx: 1.5.1
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: 1.2.0
tables: 3.3.0
numexpr: 2.6.1
matplotlib: 2.0.0
openpyxl: 2.4.1
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.2
bs4: 4.5.3
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.1.5
pymysql: None
psycopg2: None
jinja2: 2.9.4
boto: 2.45.0
pandas_datareader: 0.2.1

</details>

**Comment by jreback** [comment]
please post a copy-pastable example that reproduces the issue.

**Comment by rojour** [comment]
Well, I posted this
df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)
if that's not what you are looking for, I don't understand the question.

**Comment by jreback** [comment]
a reproducible example, one that I can simply copy-paste.

**Comment by jreback** [comment]
failing that showing ``df.info()``, ``df.head()`` might help.

**Comment by rojour** [comment]
import numpy as np
import pandas as pd

import seaborn as sns
import matplotlib.pyplot as plt
sns.set_style("whitegrid")
%matplotlib inline

from datetime import datetime
import time

parser = lambda x: pd.to_datetime(x, format='%H:%M:%S', errors='coerce')
df = pd.read_csv('marathon_results_2016.csv', date_parser=parser, parse_dates=list(range(9,21)))

s = df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)
g = sns.barplot(s.index, s, palette='Reds')
g.figure.set_size_inches(16,8)
g.set_title("Most popular City")




and you can find the entire data set here.
https://github.com/rojour/boston_results


I believe this is what your were looking for.

**Comment by jreback** [comment]
any reason you are creating non-sensical dates? These are timedeltas.

```
In [1]: import matplotlib.pyplot as plt
   ...: 

In [2]: df = pd.read_csv('t.csv')

In [3]: df.head()
Out[3]: 
  Bib                   Name  Age M/F         City State Country Citizen Unnamed: 8       5K      10K      15K      20K     Half      25K      30K      35K      40K     Pace Proj Time Official Time  \
0   5    Hayle, Lemi Berhanu   21   M  Addis Ababa   NaN     ETH     NaN        NaN  0:15:47  0:31:20  0:47:07  1:03:14  1:06:45  1:19:15  1:34:17  1:50:24  2:05:59  0:05:04   2:12:45       2:12:45   
1   1         Desisa, Lelisa   26   M         Ambo   NaN     ETH     NaN        NaN  0:15:47  0:31:21  0:47:08  1:03:14  1:06:46  1:19:15  1:34:17  1:50:24  2:05:59  0:05:06   2:13:32       2:13:32   
2   6  Tsegay, Yemane Adhane   31   M  Addis Ababa   NaN     ETH     NaN        NaN  0:15:46  0:31:20  0:47:07  1:03:13  1:06:44  1:19:15  1:34:45  1:50:48  2:06:47  0:05:07   2:14:02       2:14:02   
3  11          Korir, Wesley   33   M       Kitale   NaN     KEN     NaN        NaN  0:15:46  0:31:21  0:47:07  1:03:14  1:06:46  1:19:16  1:34:45  1:50:48  2:06:47  0:05:07   2:14:05       2:14:05   
4  14       Lonyangata, Paul   23   M      Eldoret   NaN     KEN     NaN        NaN  0:15:46  0:31:21  0:47:08  1:03:14  1:06:46  1:19:18  1:34:46  1:51:30  2:08:11  0:05:11   2:15:45       2:15:45   

   Overall  Gender  Division  
0        1       1         1  
1        2       2         2  
2        3       3         3  
3        4       4         4  
4        5       5         5  

In [4]: df.iloc[:, 9:21] = df.iloc[:, 9:21].apply(pd.to_timedelta)

In [5]: df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)
Out[5]: 
City
Boston           722
New York         451
Chicago          261
Toronto          223
San Francisco    197
Austin           178
Cambridge        177
Washington       165
Brookline        162
Houston          154
San Diego        142
Portland         137
London           133
Denver           131
Brooklyn         130
Somerville       127
Arlington        127
Seattle          122
Ottawa           110
Atlanta          103
Name: Bib, dtype: int64
```

**Comment by jreback** [comment]
maybe some interactions with ``seaborn``

```
In [1]: import matplotlib.pyplot as plt
   ...: parser = lambda x: pd.to_datetime(x, format='%H:%M:%S', errors='coerce')
   ...: df = pd.read_csv('t.csv', date_parser=parser, parse_dates=list(range(9,21)))
   ...: df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)
   ...: 
Out[1]: 
City
Boston           722
New York         451
Chicago          261
Toronto          223
San Francisco    197
Austin           178
Cambridge        177
Washington       165
Brookline        162
Houston          154
San Diego        142
Portland         137
London           133
Denver           131
Brooklyn         130
Somerville       127
Arlington        127
Seattle          122
Ottawa           110
Atlanta          103
Name: Bib, dtype: int64

In [2]: quit()
(pandas) bash-3.2$ ipython
Python 3.6.1 |Continuum Analytics, Inc.| (default, Mar 22 2017, 19:25:17) 
Type "copyright", "credits" or "license" for more information.

IPython 5.3.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import seaborn as sns
   ...: import matplotlib.pyplot as plt
   ...: parser = lambda x: pd.to_datetime(x, format='%H:%M:%S', errors='coerce')
   ...: df = pd.read_csv('t.csv', date_parser=parser, parse_dates=list(range(9,21)))
   ...: df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)
   ...: 
Bus error: 10
```

I don't rule out some pandas interactions (as we *do* setup some converters on mpl import), but think its more ilkely something else.

cc @jorisvandenbossche @tacaswell 
```
In [2]: sns.__version__
Out[2]: '0.7.1'

In [4]: import matplotlib

In [5]: matplotlib.__version__
Out[5]: '2.0.0'

In [6]: pd.__version__
Out[6]: '0.19.2'
```

**Comment by rojour** [comment]
In response to the non-sensical dates... no particular reason, however I have change them into timedeltas and it crash again.

Remove the seaborn chart, that is, I just kept the line 

s = df.groupby('City').count()['Bib'].sort_values(ascending=False).head(20)

and crash again.

Not entirely sure what to say... it's not big deal to me as I found a way around it, but I would assume that you probably want to know what's up...
I have the same versions of sns, matplotlib and pandas.
